paasta_tools/cli/cmds/validate.py

```diff
@@ -417,6 +421,58 @@ def validate_autoscaling_configs(service_path):
     return returncode
 
 
+def validate_secrets(service_path):
+    soa_dir, service = path_to_soa_dir_service(service_path)
+    system_paasta_config = load_system_paasta_config()
+    return_value = True
+    for cluster in list_clusters(service, soa_dir):
+        vault_env = system_paasta_config.get_vault_cluster_config().get(cluster)
+        if not vault_env:
+            paasta_print(failure(f"{cluster} not found on vault_cluster_map", ""))
+            return_value = False
+            continue
+
+        for instance in list_all_instances_for_service(
+            service=service, clusters=[cluster], soa_dir=soa_dir
+        ):
+            instance_config = get_instance_config(
+                service=service,
+                instance=instance,
+                cluster=cluster,
+                load_deployments=False,
+                soa_dir=soa_dir,
+            )
+            for env_value in instance_config.config_dict.get("env", {}).values():
+                if is_secret_ref(env_value):
+                    secret_name = get_secret_name_from_ref(env_value)
+                    if is_shared_secret(env_value):
+                        secret_file_name = (
+                            f"{soa_dir}/_shared/secrets/{secret_name}.json"
+                        )
+                    else:
+                        secret_file_name = f"{service_path}/secrets/{secret_name}.json"
+                    if os.path.isfile(secret_file_name):
+                        secret_json = get_config_file_dict(secret_file_name)
+                        if "ciphertext" not in secret_json["environments"].get(
+                            vault_env, {}
+                        ):
+                            paasta_print(
+                                failure(
+                                    f"Secret {secret_name} not defined for ecosystem {vault_env} on secret file {secret_file_name}",
+                                    "",
+                                )
+                            )
+                            return_value = False
+                    else:
+                        paasta_print(
+                            failure(f"Secret file {secret_file_name} not defined", "")
+                        )
+                        return_value = False
+    if return_value:
+        paasta_print(success("All secrets are present"))
```

Comment:
lets not print this message for services than don't have any secrets defined

Reply:
I think I'm going to leave this, but rephrasing the message. The extra logic needed to print based on the existence of secrets would make the function a bit more convoluted.